src/main/scala/dpla/ingestion3/harvesters/oai/OaiResponseBuilder.scala

```diff
@@ -21,95 +24,148 @@ class OaiResponseBuilder (endpoint: String)
   val urlBuilder = new OaiQueryUrlBuilder
 
   /**
-    * Get one to many pages of sets.
-    * Results will include all sets.
+    * Get one to many pages of sets, and any errors that occurred in the process
+    * of harvesting said sets.
+    *
+    * @param pageProcessor a parital function that indicates how a response from
+    *                      the OAI feed should be processed, e.g. if sets should
+    *                      be parsed according to a whitelist or blacklist.
+    *
+    * @return RDD of OaiResponses, including SetsPages and OaiErrors.
     */
-  def getSets: RDD[OaiSet] = {
+  def getSets(pageProcessor: PartialFunction[(OaiSource), OaiResponse]): RDD[OaiResponse] = {
     val baseParams = Map("endpoint" -> endpoint, "verb" -> "ListSets")
-    val response = getMultiPageResponse(baseParams)
-    val responseRdd = sqlContext.sparkContext.parallelize(response)
-    responseRdd.flatMap(page => parseSets(page))
+    val multiPageResponse = getMultiPageResponse(baseParams)
+    val responseRdd = sqlContext.sparkContext.parallelize(multiPageResponse)
+
+    responseRdd.map{
+      // Apply partial function to process the OaiSource in some specific way.
+      case page: OaiSource => pageProcessor(page)
+      // Retain other responses (ie. errors) as given.
+      case response: OaiResponse => response
+    }
+  }
+
+  def getAllSets(): RDD[OaiResponse] = {
+    val allSetsPf: PartialFunction[(OaiSource), OaiResponse] = {
+      case(page) => OaiResponseProcessor.getAllSets(page)
```

Comment:
Not familiar with this `case(blah)` syntax, is that different than `case blah:`?